### PR TITLE
Fixes throw impact doing twice the normal damage

### PR DESCRIPTION
**Pull request by yoyobatty**
[Guidelines]: # (Be sure that your PR follows our guidelines, such as modularization and comment standards. You can read more about the subject here: https://github.com/HippieStation/HippieStation/blob/master/hippiestation/README.md )

[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs )

:cl: YoYoBatty
fix: Impacting a wall no longer does 20 damage
/:cl:

[why]: Big fucking oof coders.

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -93,7 +93,11 @@
 					return 1
 	return ..()
 
-/mob/living/carbon/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+/mob/living/carbon/proc/reset_fist_casted()//hippie edit -- adds plasma fist
+	if(fist_casted)
+		fist_casted = FALSE //hippie end -- adds plasma fist
```
Comment (GuyonBroadway) [inline code/modules/mob/living/carbon/carbon.dm]: It's not plasma fist you dweebus its just "fist"

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -93,7 +93,11 @@
 					return 1
 	return ..()
 
-/mob/living/carbon/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+/mob/living/carbon/proc/reset_fist_casted()//hippie edit -- adds fist
```
Comment (GuyonBroadway) [inline code/modules/mob/living/carbon/carbon.dm]: This can be modularized as its a new proc

**Review by GuyonBroadway — DISMISSED** [review]

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -106,6 +106,11 @@
 		if(hurt)
 			Paralyze(20)
 			take_bodypart_damage(10)
+		if(fist_casted)//hippie edit -- adds fist
+			var/turf/T = get_turf(src)
+			visible_message("<span class='danger'>[src] slams into [T] with explosive force!</span>", "<span class='userdanger'>You slam into [T] so hard everything nearby feels it!</span>")
+			explosion(T, -1, 1, 4, 0, 0, 0) //No fire and no flash, this is less an explosion and more a shockwave from beign punched THAT hard.
+			fist_casted = FALSE //hippie end -- fist
```
Comment (Carbonhell) [inline code/modules/mob/living/carbon/carbon.dm:113]: If you just set this to false, what is the point of the /reset_fist_casted() proc?

**Review by Carbonhell — CHANGES_REQUESTED** [review]

**Review by Carbonhell — DISMISSED** [review]

**Review by Carbonhell — APPROVED** [review]

**Comment by Jujumatic** [comment]
stop making branches on our FUCKING REPO REEEEEEEEEEEEEEEEEEEEE

**Pull request merged**